Shelf.cs

```diff
@@ -0,0 +1,208 @@
+                    _freeSpace = value;
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine("שגיאה: " + e.Message);
+                }
+
+            }
+        }
+        public int CurrentFreeSpace
+        {
+            get
+            { return _currentFreeSpace; }
+
+            set
+            {
+                try
+                {
+                    if (value <= 0) throw new Exception("invalide size");
+                    _currentFreeSpace = value;
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine("שגיאה: " + e.Message);
+                }
+            }
+        }
+        public int GetCurrentFreeSpace()
+        {
+            return CurrentFreeSpace;
+        }
+
+        public bool AddItemToShelf(Item item, Shelf shelf)
+        {
+            if (item.Size <= this.GetCurrentFreeSpace())
+            {
+                item.Shelf = shelf;
+                this.Items.Add(item);
+                this.CurrentFreeSpace -= item.Size;
+                return true;
+
+            }
+            return false;
+        }
+
+        //4
+        public Item SearchItemOnItems(int itemId)
+        {
+            foreach (Item item in Items)
+            {
+                if (item.Id == itemId)
+                {
+                    return item;
+                }
+            }
+            return null;
+        }
+
+        //4
+        public Item RemoveItemFromShelf(int itemId)
+        {
+            Item item = SearchItemOnItems(itemId);
+            if (item != null)
+            {
+                item.Shelf = null;
+                this.Items.Remove(item);
+                this.CurrentFreeSpace += item.Size;
+            }
+
+            return item;
+        }
+
+        public void RemoveItemByKashrutAndDate(Kashrut kashrut, int countDay)
+        {
+            DateTime currentTime = DateTime.Now;
+            DateTime ExpirationDateFinall = currentTime.AddDays(countDay);
+            List<Item> filteredNumbers = new List<Item>();
+
+            foreach (Item item in Items)
+            {
+                if (item.Kashrut == kashrut && item.ExpirationDate < ExpirationDateFinall)
+                    RemoveItemFromShelf(item.Id);
+            }
+
+        }
+
+        //4
+        public Item REmoveItem1(Item item)
+        {
+            item.Shelf = null;
+            this.Items.Remove(item);
+            this.CurrentFreeSpace += item.Size;
+            return item;
+        }
+
+        public String CleanExpiredFromShelf()
+        {
+            DateTime currentTime = DateTime.Now;
+            String itemlist = "";
+            foreach (Item item in Items)
+            {
+                itemlist = item.ToString() + "\n";
+                if (item.ExpirationDate < currentTime)
+                {
+                    REmoveItem1(item);
+                }
+
+            }
+
```

Comment:
Bug: everytime this function run, it crashes the whole system. This is because you change an array while iterating it, which makes the iteration invalid. Please stop doing this, thank you